blueocean-commons/src/main/java/io/jenkins/blueocean/commons/analytics/Analytics.java

```diff
@@ -0,0 +1,70 @@
+package io.jenkins.blueocean.commons.analytics;
+
+import com.google.common.base.Objects;
+import com.google.common.collect.Maps;
+import com.google.common.hash.Hashing;
+import hudson.model.UsageStatistics;
+import hudson.model.User;
+import io.jenkins.blueocean.commons.ServiceException;
+import org.apache.commons.lang.StringUtils;
+import org.jenkinsci.main.modules.instance_identity.InstanceIdentity;
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
+import org.kohsuke.stapler.DataBoundConstructor;
+
+import java.util.Map;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+@Restricted(NoExternalUse.class)
+public abstract class Analytics {
```

Comment:
Maybe where I was going was that, how could other plugins participate in sending their events as well. Not immediate need but some extensions that could let them do it. For example, what if pipeline editor wants to add it's events, e.g. editor version etc.